e2e/consulcompat/input/consul-policy-for-nomad.hcl

```diff
@@ -14,15 +14,11 @@ agent_prefix "" {
   policy = "read"
 }
 
-key_prefix "" {
-  policy = "read"
-}
-
 node_prefix "" {
   policy = "read"
 }
 
-service_prefix "" {
+service_prefix "nomad" {
```

Comment:
Since the Nomad agent runs in `dev` mode, this policy is used for both client and  servers, so this will prevent clients from registering in Consul and also from syncing services.
https://github.com/hashicorp/nomad/blob/5ad715b2818041674645486862415848dc425f48/command/agent/consul/service_client.go#L984-L988

If we don't care about gateways and their config entries, I think the most restrictive policy we can have that is fully functionable would be:

```hcl
service_prefix "nomad" {
  policy = "write"
}

service_prefix "nomad-client" {
  policy = "write"
}

service_prefix "" {
  policy = "read"
}
```

Reply:
`service_prefix` is a prefix match, so I think we can collapse the top two